src/Servers/Kestrel/Core/src/Internal/Infrastructure/TimingPipeFlusher.cs

```diff
@@ -51,35 +51,39 @@ public ValueTask<FlushResult> FlushAsync(MinDataRate minRate, long count)
 
         public ValueTask<FlushResult> FlushAsync(MinDataRate minRate, long count, IHttpOutputAborter outputAborter, CancellationToken cancellationToken)
         {
-            var flushValueTask = _writer.FlushAsync(cancellationToken);
-
-            if (minRate != null)
-            {
-                _timeoutControl.BytesWrittenToBuffer(minRate, count);
-            }
-
-            if (flushValueTask.IsCompletedSuccessfully)
-            {
-                return new ValueTask<FlushResult>(flushValueTask.Result);
-            }
-
             // https://github.com/dotnet/corefxlab/issues/1334
             // Pipelines don't support multiple awaiters on flush.
-            // While it's acceptable to call PipeWriter.FlushAsync again before the last FlushAsync completes,
-            // it is not acceptable to attach a new continuation (via await, AsTask(), etc..). In this case,
-            // we find previous flush Task which still accounts for any newly committed bytes and await that.
             lock (_flushLock)
             {
-                if (_lastFlushTask == null || _lastFlushTask.IsCompleted)
+                if (_lastFlushTask != null && !_lastFlushTask.IsCompleted)
```

Comment:
`.IsCompletedSuccessfully`?

Want Cancellation and Exceptions to go down the awaited path? (i.e. to throw and wrap their exceptions, rather than ignore them)

Reply:
nvm, see what's happening

Reply:
Yeah. I was thinking about this too, but my reasoning for not awaiting any completed Task is that if it is a persistent error, the next Flush should return the same error anyway. If not, all the better.

Reply:
The remaining question I had is whether to swallow any exceptions that might be thrown from an awaited _lastFlushTask. Right now, I don't.